src/diffusion_for_multi_scale_molecular_dynamics/models/score_networks/mlp_score_network.py

```diff
@@ -39,27 +48,37 @@ def __init__(self, hyper_params: MLPScoreNetworkParameters):
             hyper_params : hyper parameters from the config file.
         """
         super(MLPScoreNetwork, self).__init__(hyper_params)
-        hidden_dimensions = [
-            hyper_params.hidden_dimensions_size
-        ] * hyper_params.n_hidden_dimensions
+        hidden_dimensions = [hyper_params.hidden_dimensions_size] * (
+            hyper_params.n_hidden_dimensions
+        )
         self._natoms = hyper_params.number_of_atoms
-
-        output_dimension = self.spatial_dimension * self._natoms
-        input_dimension = output_dimension + hyper_params.embedding_dimensions_size
+        self.num_atom_types = hyper_params.num_atom_types
+
+        coordinate_output_dimension = self.spatial_dimension * self._natoms
+        atom_type_output_dimension = self._natoms * (self.num_atom_types + 1)
+        input_dimension = (
+            coordinate_output_dimension
+            + hyper_params.noise_embedding_dimensions_size
+            + self._natoms * hyper_params.atom_type_embedding_dimensions_size
+        )
 
         self.noise_embedding_layer = nn.Linear(
-            1, hyper_params.embedding_dimensions_size
+            1, hyper_params.noise_embedding_dimensions_size
+        )
+
+        self.atom_type_embedding_layer = nn.Linear(
+            self.num_atom_types + 1, hyper_params.atom_type_embedding_dimensions_size
```

Comment:
create a variable `number_of_classes` instead of using `self.num_atom_types + 1`.